### PR TITLE
[Core]: Update to v2 spec

### PR DESCRIPTION
**Pull request by meatball133**
Updated dependencies:
- Node 14 > Node 12
- CoffeeScript 2.7 > 2.5.1

### CODE REVIEW

File: tests/example-syntax-error/expected_results.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
-  "version": 1,
-  "status": "fail",
-  "message": "Exception loading: /solution/example-syntax-error.spec.coffee\n[/solution/example-syntax-error.coffee:1:13: error: unexpected @\nclasSSY Bo%b@\n            ^] {\n  location: { first_line: 0, first_column: 12, last_line: 0, last_column: 12 },\n  toString: [Function: syntaxErrorToString],\n  code: 'clasSSY Bo%b@',\n  filename: '/solution/example-syntax-error.coffee'\n}\nunexpected @"
-}
+	"version": 2,
+	"status": "fail",
```
Comment (ErikSchierboom) [inline tests/example-syntax-error/expected_results.json:3]: This should technically be an `error` status, as the test could not run.

File: tests/example-empty-file/expected_results.json [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
-}
+	"version": 2,
+	"status": "fail",
+	"tests": [
+		{
+			"name": "encountered a declaration exception",
+			"status": "fail",
+			"message": "TypeError: Bob is not a constructor"
+		}
```
Comment (ErikSchierboom) [inline tests/example-empty-file/expected_results.json:9]: This seems not to be the actual test but the error.

Comment (meatball133) [reply tests/example-empty-file/expected_results.json:9]: Yeah, it is because the error doesn't occur in a test case but actually outside the test case

Comment (ErikSchierboom) [reply tests/example-empty-file/expected_results.json:9]: Yes. Can you convert this to a top-level `error` with no tests?

File: Dockerfile [hunk 3]
```diff
@@ -1,19 +1,11 @@
...
+RUN apk add --no-cache jq coreutils bash
 
 ENV NO_UPDATE_NOTIFIER true
 
 WORKDIR /opt/test-runner
+COPY . .
 
-# Pre-install packages
-COPY package.json .
-COPY package-lock.json .
```
Comment (ErikSchierboom) [inline Dockerfile:15]: Why did you remove this bit of the Dockerfile? This is supposed to make better use of Docker layer caching as those two files are less likely to change than the other files

**Review by ErikSchierboom — COMMENTED** [review]
One tiny comment

File: Dockerfile (hunk 3 above)
Comment (meatball133) [reply Dockerfile:15]: Since I think those get copied when doing COPY . .

Comment (ErikSchierboom) [reply Dockerfile:15]: Yes, but the idea is that you copy just those files first, then do the `npm install` and after that do the `COPY . .`. This has the effect of docker caching the NPM install layer if these two files don't change. And as the NPM install step is relatively quite long, this can speed up builds.

**Review by ErikSchierboom — APPROVED** [review]
